fog/view/server/src/router_request_handler.rs

```diff
@@ -27,7 +28,7 @@ const RETRY_COUNT: usize = 3;
 /// Handles a series of requests sent by the Fog Router client.
 pub async fn handle_requests<E>(
     method_name: GrpcMethodName,
-    shard_clients: Vec<Arc<FogViewStoreApiClient>>,
+    shards: Vec<Shard>,
```

Comment:
This is doing two additional `Clone`s compared to the previous, cloning a URI and a block range. I believe this should be cheap, so I'm not too worried, but the cost of cloning a URI should at least be looked at.

Reply:
Ack.